tools/swiftdt/main.swift

```diff
@@ -0,0 +1,166 @@
+import SwiftRemoteMirror
+
+
+func argFail(_ message: String) -> Never {
+  print(message, to: &Std.err)
+  exit(EX_USAGE)
+}
+
+func machErrStr(_ kr: kern_return_t) -> String {
+    let errStr = String(cString: mach_error_string(kr))
+    let errHex = String(kr, radix: 16)
+    return "\(errStr) (0x\(errHex))"
+}
+
+
+var argv = ArraySlice(CommandLine.arguments)
+guard let executableName = argv.popFirst() else {
+  argFail("Command line arguments are completely empty!")
+}
+
+struct Command {
+  var name: String
+  var help: String
+  var call: (inout ArraySlice<String>) throws -> Void
+  
+  init(name: String, help: String,
+       call: @escaping (inout ArraySlice<String>) -> Void) {
+    self.name = name
+    self.help = help
+    self.call = call
+  }
+  
+  init(name: String, help: String,
+       call: @escaping (SwiftReflectionContextRef) throws -> Void) {
+    self.name = name
+    self.help = help
+    self.call = { try withReflectionContext(args: &$0, call: call) }
+  }
+}
+
+let commands = [
+  Command(
+    name: "dump-conformance-cache",
+    help: "Print the contents of the target's protocol conformance cache.",
+    call: dumpConformanceCache),
+  Command(
+    name: "dump-metadata-allocations",
+    help: "Print the target's metadata allocations.",
+    call: dumpMetadataAllocations),
+  Command(
+    name: "help",
+    help: "Print this help.",
+    call: printUsage),
+]
+
+func dumpConformanceCache(context: SwiftReflectionContextRef) throws {
+  try context.iterateConformanceCache(call: { type, proto in
+    let typeName = context.name(metadata: type) ?? "<unknown>"
+    let protoName = context.name(proto: proto) ?? "<unknown>"
+    print("Conformance: \(typeName): \(protoName)")
+  })
+}
+
+func dumpMetadataAllocations(context: SwiftReflectionContextRef) throws {
+  var allocations: [swift_metadata_allocation_t] = []
+  var metadatas: [swift_reflection_ptr_t] = []
+  try context.iterateMetadataAllocations(call: { allocation in
+    allocations.append(allocation)
+    print("Metadata allocation at: \(hex: allocation.Ptr) " + 
+          "size: \(allocation.Size) tag: \(allocation.Tag)")
+    let ptr = context.metadataPointer(allocation: allocation)
+    if ptr != 0 {
+      metadatas.append(ptr)
+    }
+  })
+
+  allocations.sort(by: { $0.Ptr < $1.Ptr })
+  for metadata in metadatas {
+    let name = context.name(metadata: metadata) ?? "<unknown>"
+    print("Metadata \(hex: metadata)")
+    print("    Name: \(name)")
+
+    if let allocation = allocations.last(where: { metadata >= $0.Ptr }) {
+      let offset = metadata - allocation.Ptr
+      print("    In allocation \(hex: allocation.Ptr) " +
+            "size \(allocation.Size) at offset \(offset)")
+    } else {
+      print("    Not in any known metadata allocation. How strange.")
+    }
+  }
+}
+
+func printUsage(args: inout ArraySlice<String>) {
+  print("Usage: \(executableName) <command>", to: &Std.err)
```

Comment:
`print("Usage: \(executableName) <pid | process name> <command>", to: &Std.err)`

Reply:
@gottesmm helped me transform this into a proper package, so I went ahead and redid it to use `ArgumentParser` instead of rolling my own. Now we get all the right behavior for free!